packages/trader/src/App/Components/Elements/PositionsDrawer/positions-drawer.jsx

```diff
@@ -5,64 +5,215 @@ import React                          from 'react';
+        if (this.list_ref && this.hasPositionsHeightChanged(newPositionsHeight, this.positionsHeight)) {
+            // When there is a change in height of an item, this recalculates scroll height of the list and reapplies styles.
+            setTimeout(() => this.list_ref ? this.list_ref.resetAfterIndex(0) : undefined);
+        }
+
+        this.positionsHeight = newPositionsHeight;
+    }
+
+    getCachedPositionHeight = (index) => {
+        return this.positionsHeight[index];
+    }
+
+    getPositionHeight = (position) => {
+        // React window doesn't work with dynamic height. This is a work around to get height of a position based on different combinations.
+        const { contract_info } = position;
+        const has_ended = isEnded(contract_info);
+        const is_valid_to_sell   = isValidToSell(contract_info);
+        const is_multiplier_contract = isMultiplierContract(contract_info.contract_type);
+        const is_tick_contract = contract_info.tick_count > 0;
+
+        if (has_ended) {
+            return is_multiplier_contract ? 198 : 158;
+        } else if (is_tick_contract) {
+            return 202;
+        }
+        
+        const classic_contract_height = is_valid_to_sell ? 228 : 188;
+        return is_multiplier_contract ? 238 : classic_contract_height;
+    }
+
+    getListScrollbar() {
+        const ListScrollbar = React.forwardRef((props, ref) => {
+            const { children, style, onScroll } = props;
+
+            const refCallback = (forwardRef) => {
+                this.scrollbar_ref = forwardRef;
+                ref.call(this, forwardRef);
+            };
+
+            return (
+                <ThemedScrollbars
+                    list_ref={refCallback}
+                    style={{ ...style, overflow: 'hidden' }}
+                    onScroll={onScroll}
+                    autoHide
+                >
+                    {children}
+                </ThemedScrollbars>
+            );
+        });
+        // Display name is required by Developer Tools to give a name to the components we use.
+        // If a component doesn't have a displayName is will be shown as <Unknown />. Hence, name is set.
+        ListScrollbar.displayName = 'ListScrollbar';
+
+        return ListScrollbar;
+    }
+
     render() {
         const {
             all_positions,
             error,
-            currency,
             is_empty,
             is_positions_drawer_on,
-            onClickSell,
-            onClickRemove,
+            symbol,
             toggleDrawer,
-            toggleUnsupportedContractModal,
         } = this.props;
 
-        // Show only 5 most recent open contracts
+        this.positions = all_positions.filter((p) => (
+            p.contract_info && symbol === p.contract_info.underlying) &&
+            this.filterByContractType(p.contract_info)
+        );
+        this.calculatePositionsHeight();
```

Comment:
Not a huge fan of this as it will be 1. difficult to maintain and 2.  won't work with translations and 3. I think performance of react-virtualized s better for calculating heights.

but @balakrishna-binary mentioned that the list will change to `react-virtualized` in a future PR to handle dynamic heights so I think it's ok for now 😄

Reply:
Thanks @oskar-binary. Will change it soon 😃